orthoinference/src/test/java/org/reactome/orthoinference/PathwayDiagramGeneratorTest.java

```diff
@@ -0,0 +1,81 @@
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+//@RunWith(PowerMockRunner.class)
+@PrepareForTest({PathwayDiagramGenerator.class})
+@PowerMockIgnore({"org.apache.logging.log4j.*", "javax.management.*", "javax.script.*",
+        "javax.xml.*", "com.sun.org.apache.xerces.*", "org.xml.sax.*", "com.sun.xml.*", "org.w3c.dom.*", "org.mockito.*"})
+public class PathwayDiagramGeneratorTest {
+
+    @Mock
+    MySQLAdaptor mockAdaptor;
+    @Mock
+    GKInstance mockSpeciesInst;
+    @Mock
+    GKInstance mockDiagramInst;
+    @Mock
+    GKInstance mockPathwayInst;
+    @Mock
+    GKInstance mockOrthoEventInst;
+    @Mock
+    GKInstance mockEvidenceTypeInst;
+    @Mock
+    GKInstance mockOrthoDiagramInst;
+    @Mock
+    PredictedPathwayDiagramGeneratorFromDB mockDiagramGenerator;
+
+    Collection<GKInstance> mockDiagramInstances = new ArrayList<>();
+    List<GKInstance> mockOrthoEventInstances = new ArrayList<>();
+
+
+    PathwayDiagramGenerator testDiagramGenerator;
+
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        testDiagramGenerator = new PathwayDiagramGenerator();
+    }
+    @Test
+    public void generateOrthologousPathwayDiagramsTest() throws Exception {
+
+        mockDiagramInstances.add(mockDiagramInst);
+        mockOrthoEventInstances.add(mockOrthoEventInst);
+
+        PathwayDiagramGenerator.setAdaptor(mockAdaptor);
+        PathwayDiagramGenerator.setSpeciesInstance(mockSpeciesInst);
+        PathwayDiagramGenerator.setReferenceSpeciesId(12345L);
+        PathwayDiagramGenerator.setPersonId(12345L);
+
+        Mockito.when(mockAdaptor.fetchInstance(12345L)).thenReturn(mockSpeciesInst);
+        Mockito.when(mockSpeciesInst.getDisplayName()).thenReturn("Species");
+        Mockito.when(mockAdaptor.fetchInstancesByClass(ReactomeJavaConstants.PathwayDiagram)).thenReturn(mockDiagramInstances);
+        Mockito.when(mockDiagramInst.getAttributeValue(ReactomeJavaConstants.representedPathway)).thenReturn(mockPathwayInst);
+        Mockito.when(mockPathwayInst.getAttributeValue(ReactomeJavaConstants.species)).thenReturn(mockSpeciesInst);
+        Mockito.when(mockPathwayInst.getAttributeValuesList(ReactomeJavaConstants.orthologousEvent)).thenReturn(mockOrthoEventInstances);
+        Mockito.when(mockOrthoEventInst.getAttributeValue(ReactomeJavaConstants.species)).thenReturn(mockSpeciesInst);
+        Mockito.when(mockOrthoEventInst.getAttributeValue(ReactomeJavaConstants.evidenceType)).thenReturn(mockEvidenceTypeInst);
+//        Mockito.when(mockDiagramGenerator.generatePredictedDiagram(mockOrthoEventInst, mockPathwayInst, mockDiagramInst)).thenReturn(mockOrthoDiagramInst);
+//
+        PathwayDiagramGenerator.generateOrthologousPathwayDiagrams();
```

Comment:
Is there a way to assert that the diagrams were actually generated correctly? Maybe if `generateOrthologousPathwayDiagrams` returned some value, such as the number of diagrams that were generated, then this would be easier to verify?

Reply:
That would be quite nice, but I think since it is all mocked objects I would need a way to check that doesn't involve accessing a GKInstance object. I'll look into a better way of testing this so there is some assertion that can be made.